src/server/scene/surface_stack.cpp

```diff
@@ -363,15 +443,40 @@ void ms::SurfaceStack::update_rendering_tracker_compositors()
         pair.second->active_compositors(registered_compositors);
 }
 
+void ms::SurfaceStack::insert_surface_at_top_of_depth_layer(std::shared_ptr<Surface> const& surface)
+{
+    unsigned int depth_index = mir_depth_layer_to_index(surface->depth_layer());
+    if (surface_layers.size() <= depth_index)
+        surface_layers.resize(depth_index + 1);
+    surface_layers[depth_index].push_back(surface);
+}
+
+auto ms::SurfaceStack::mir_depth_layer_to_index(MirDepthLayer depth_layer) -> unsigned int
+{
+    switch (depth_layer)
+    {
+    case mir_depth_layer_background:        return 0;
+    case mir_depth_layer_below:             return 1;
+    case mir_depth_layer_application:       return 2;
+    case mir_depth_layer_always_on_top:     return 3;
+    case mir_depth_layer_above:             return 4;
+    case mir_depth_layer_overlay:           return 5;
+    }
+    assert(false);
```

Comment:
Clang will assume that the `depth_layer` parameter contains a valid enum value, will happily accept this code without the `assert()`, and *without* the `assert()` will check that the switch above is exhaustive.

[This](https://github.com/MirServer/mir/blob/954d759c552f53730d1a725fd29a815ea45c43b8/src/common/posix_rw_mutex.cpp#L81) provides an example of how to do this.

Reply:
With a default case I don't get a warning/error if I'm missing an enum case (which I do with GCC now). I would hope Clang checks that the switch is exhaustive  with or without the assert.

Reply:
I added the `#ifndef __clang__`, but not in a default case.

Reply:
Aha! This must be a new thing for GCC. Cool!